File: programming files/even_odd_site.cc
// bugs introduced: ws
#include <iostream>
#include <fstream>
using namespace std;

int main() {
    // Open file stream
    ofstream file;
    // numbers.html is opened file for writing
    file.open("numbers.html");
    // Add head/title tags and opening body tag to html
    file << "<html>\n<head>\n<title>List of Numbers</title>\n</head>\n<body>\n";
    // Write table headers
    file << "<table>\n<tr><th>Even Numbers</th><th>Odd Numbers</th></tr>\n";
    // Write numbers from 1 to 50 in table
    for (int i = 1; i < 50; i++) {
        // If even number
        if (i % 2 == 1) {
            file << "<tr><td>" << i << "</td><td></td></tr>\n";
        }
        // If odd number
        else {
            file << "<tr><td></td><td>" << i << "</td></tr>\n";
        }
    
    // Close table, body, html tags
    file << "</table>\n</body>\n<html>";
    // Close file stream
    file.close();
    // Open filestream to read file
    ifstream input("numbers.html");
    // Print file info
    cout << input.rdbuf()
    // Close file stream
    input.close();
    return 0;
}

File: programming files/even_odd_site.py
#broken by: ws
#open numbers.html
with open("numbers.html", "w") as f:
    #add the title
    f.write("<html>\n<head>\n<title>List of Numbers</title>\n</head>\n<body>\n")
    #add the table header
    f.write("<table>\n<tr><th>Even Numbers</th><th>Odd Numbers</th></tr>\n")
    #loop through the numbers 1 to 50
    for i in range(1, 51):
        #check if even or odd
        if i % 2 == 0:
            #write to the even column
            f.write("<tr><td>{}</td><td></td></tr>\n".format(i+1))
        else:
            #write to the odd column
            f.write("<tr><td></td><td>{}</td></tr>\n".format(i+1))
    #close the headers
    f.write("</table>\n</body>\n</html>")

#display the html code
with open("numbers.html", "r") as f:
    print(f.read())
    
